src/redux/books/books.js

```diff
@@ -1,44 +1,38 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getBooks, deleteBook, postBook } from '../../API/bookapi';
+
 const ADD_BOOK = 'book-store-react/books/ADD_BOOK';
+const FETCH_BOOKS = 'book-store-react/books/FETCH_BOOKS';
 const REMOVE_BOOK = 'book-store-react/books/REMOVE_BOOK';
 
-// Reducer
-const booksReducer = (state = [], action) => {
-  switch (action.type) {
-    case ADD_BOOK:
-      return [
-        ...state,
-        {
-          id: action.id,
-          title: action.title,
-          author: action.author,
-        },
-      ];
-
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.id);
-
-    default:
-      return state;
-  }
-};
-
-// actions
-export const addBook = (bookTitle, bookAuthor, bookId) => {
-  const book = {
-    type: ADD_BOOK,
-    title: bookTitle,
-    author: bookAuthor,
-    id: bookId,
-  };
-  return book;
-};
+// fetch books from the api
+export const fetchBooks = createAsyncThunk(FETCH_BOOKS,
+  async () => {
+    const response = await getBooks();
+    return response;
+  });
+// add book to api
+export const addBook = createAsyncThunk(ADD_BOOK,
+  async (bookdetails) => {
+    const response = await postBook(bookdetails);
+    return response;
+  });
```

Comment:
- [ ] After successfully creating the book using the POST request, you should update the books state to reflect the addition of a new book. That means, you need to dispatch an actions that updates the list of books in the store.